File: Frontend/proyecto/src/types/Message.ts
import { User } from "./User";
import { Chat } from "./Chat";

export interface Message {
  id: string;
  chatId: string;
  senderId: string;
  senderName: string;
  content: string;
  sentAt: string;
  sender?: User;   // opcional, si el backend lo incluye
  chat?: Chat;     // opcional
  timestamp?: string;
}

File: Catalog/Models/Message.cs


namespace Catalog.Models
{
    public class Message
    {
        public Guid Id { get; set; }
        public Guid ChatId { get; set; }
        public Guid SenderId { get; set; }
        public string Content { get; set; } = null!;
        public DateTime SentAt { get; set; } = DateTime.UtcNow;

        // Relaciones
        public Chat Chat { get; set; } = null!;
    }

}
